python/tank/platform/environment.py

```diff
@@ -878,3 +898,284 @@ def copy_apps(self, src_engine_name, dst_engine_name):
         self.__write_data(self._env_path, data)
         # sync internal data with disk
         self._refresh()
+
+
+    ############################################################################
+    # Methods specific to dumping environment settings
+
+    def dump(self, file):
+        """
+        Dump a copy of this environment's settings to a file.
+
+        NOTE: The calling code is responsible for closing the file handle.
+
+        :param file: An open file object to write to.
+        """
+
+        # load the output path's yaml
+        yml_data = self.__load_writable_yaml(self._env_path)
+
+        try:
+            self.__write_data_file(file, yml_data)
+        except Exception, e:
+            raise TankError(
+                "Could not write to environment file handle. "
+                "Error reported: %s" % (e,)
+            )
+
+    def dump_sparse(self, file, debug=False):
+        """
+        Dump a sparse copy of this environment's settings to a file.
+
+        Any setting defined in the environment that matches the default value
+        specified will not be included.
+
+        NOTE: The calling code is responsible for closing the file handle.
+
+        :param file: An open file object to write to.
+        :param debug: If True, include debug comments on lines using a
+            non-default value.
+        """
+
+        self._dump_settings(self.SPARSE, file, debug)
+
+    def dump_full(self, file, debug=False):
+        """
+        Dump a full copy of this environment's settings to a file.
+
+        Any setting not specified in the environment will be dumped with the
+        default value.
+
+        NOTE: The calling code is responsible for closing the file handle.
+
+        :param file: An open file object to write to.
+        :param debug: If True, include debug comments on lines using a default
+            value.
+        """
+
+        self._dump_settings(self.FULL, file, debug)
+
+    def _update_settings_sparse(self, schema, settings, engine_name=None,
```

Comment:
at first glance, this method looks similar to `_update_settings_full` - would it make sense to combine them?